sdk/src/logs/batch_log_record_processor.cc

```diff
@@ -110,35 +114,17 @@ bool BatchLogRecordProcessor::ForceFlush(std::chrono::microseconds timeout) noex
   bool result = false;
   while (!result && timeout_steady > std::chrono::steady_clock::duration::zero())
   {
-    // When is_force_flush_notified.store(true) and force_flush_cv.notify_all() is called
-    // between is_force_flush_pending.load() and force_flush_cv.wait(). We must not wait
-    // for ever
+    // When force_flush_notified_sequence.compare_exchange_strong(...) and
+    // force_flush_cv.notify_all() is called between force_flush_pending_sequence.load(...) and
+    // force_flush_cv.wait(). We must not wait for ever
     std::chrono::steady_clock::time_point start_timepoint = std::chrono::steady_clock::now();
     result = synchronization_data_->force_flush_cv.wait_for(lk_cv, scheduled_delay_millis_,
```

Comment:
If the value of `timeout` is lesser than that of `scheduled_delay_millis_`, then we'll wait longer than expected I guess

Reply:
Good point, it's fixed now.